sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -659,3 +659,25 @@ def test_early_stopping_on_test_set_with_warm_start():
     # does not raise on second call
     gb.set_params(max_iter=2)
     gb.fit(X, y)
+
+
+@pytest.mark.parametrize('Est', (HistGradientBoostingClassifier,
+                                 HistGradientBoostingRegressor))
+def test_single_node_trees(Est):
+    # Make sure it's still possible to build single-node trees. In that case
+    # the value of the root is set to 0. That's a correct value: if the tree is
+    # single-node that's because min_gain_to_split is not respected right from
+    # the root, so we don't want the tree to have any impact on the
+    # predictions.
+
+    X, y = make_classification(random_state=0)
+    y[:] = 1  # constant target will lead to a single root node
+
+    est = Est(max_iter=20)
+    est.fit(X, y)
+
+    assert all(len(predictor[0].nodes) == 1 for predictor in est._predictors)
+    assert all(predictor[0].nodes[0]['value'] == 0
+               for predictor in est._predictors)
```

Comment:
Note that in master, the value of the root would effectively be the same: the gradients for that tree would all be zeros, because the baseline prediction gives a perfect prediction already.
So that would also lead to a 0 value (even though the value would have been computed from the formula, instead of being hard-coded).

So the change preserves backward  compatibility.

Reply:
nice! Thanks :)